src/shared/tpm2-util.c

```diff
@@ -256,6 +256,25 @@ static bool tpm2_test_parms(Tpm2Context *c, TPMI_ALG_PUBLIC alg, const TPMU_PUBL
         return rc == TSS2_RC_SUCCESS;
 }
 
+static inline bool tpm2_supports_tpmt_sym_def_object(Tpm2Context *c, const TPMT_SYM_DEF_OBJECT *parameters) {
+        assert(parameters);
+
+        return tpm2_test_parms(c, TPM2_ALG_SYMCIPHER, &(const TPMU_PUBLIC_PARMS){ .symDetail.sym = *parameters, });
+}
+
+static inline bool tpm2_supports_tpmt_sym_def(Tpm2Context *c, const TPMT_SYM_DEF *parameters) {
+        assert(parameters);
+
+        /* Unfortunately, TPMT_SYM_DEF and TPMT_SYM_DEF_OBEJECT are separately defined, even though they are
+         * functionally identical. */
+        const TPMT_SYM_DEF_OBJECT object = {
+                .algorithm = parameters->algorithm,
+                .keyBits = parameters->keyBits,
+                .mode = parameters->mode,
+        };
+        return tpm2_supports_tpmt_sym_def_object(c, &object);
```

Comment:
doesn't matter, but you could also make this a compound initializer, i.e. have no local named variable at all

Reply:
I'm fine either way, is that the convention/preference?